### ISSUE TITLE
Should not be able to restore an anonymous (pure) stub

### ISSUE THREAD

**Issue by jonathan-benn** [note35/sinon#44]
For anonymous (pure) stubs, that is to say a stub created via `stub = sinon.stub()` that is not overriding another function, the **Sinon.JS** library does not allow you to call the `restore()` method.

_e.g._

```Javascript
stub = sinon.stub()
stub.restore() // throws "TypeError: stub.restore is not a function"
```

While the exact error message doesn't matter too much, **Sinon.PY** should probably also throw an error. It doesn't make sense to try to restore a pure stub.

**Comment by note35** [comment]
Thanks for pointing out this.

However, based on current design, getting rid of `wrap()`, `_queue` from `anonymous stub` will impact many things. Especially for code consistency, thus, I prefer to keep the current design until other more important issues are fixed.

**Comment by jonathan-benn** [comment]
I don't think it would be necessary to change the design. This is a simple change.

_e.g._

```python
def restore(self):
    ...
    if self.args_type is "PURE":
        raise Exception( ... )
    ...
```

**Comment by note35** [comment]
The solution is simple but unsafe.

If user try to write:

```python
stub = sinon.stub()
try:
    stub.restore() // throws "TypeError: stub.restore is not a function"
except:
    ...
# later, the _queue still contains anonymous stub, it should be fixed
# clean up `pure` might be harmless in this case, but it will make code inconsistent against to other types
```

And to be honest, there are many code for pure only, for fixing this, it's better to get rid of them to.
